### PR TITLE
Include list of defaults and example usage in perf help. #719

### LINKED ISSUE #719: [mesheryctl perf] Include list of defaults and example usage in `perf` help
**Current Behavior**
While `perf` doesn’t mandate users to populate all (optional) flags with values (it conveniently supplies defaults for these optional flags), users may not be aware of what those defaults are. 

**Desired Behavior**
When a user executes `mesheryctl perf` or `mesheryctl perf --help`, the command should output help text, including default values for flags along with example usage. 

When `mesheryctl perf` is executed without all required flags, it should return the help text output and not take any other action (like generating a test name).

See resources below for reference. 

---
**Resources**
- [Meshery CLI Commands & Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#)

### PR DESCRIPTION
**Pull request by chandresh-pancholi**
Signed-off-by: chandresh-pancholi <chandreshpancholi007@gmail.com>

#### Description

This PR fixes #719     

#### Checklist

- [] [DCO](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin) signed.
- [] Title of the PR starts with area name, e.g. `meshery:`, `mesheryctl:`, `ui:`.

### CODE REVIEW

File: mesheryctl/cmd/perf.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ var (
 	testMesh = ""
 	// testFile         = ""
 	qps              = ""
-	concurrentRequests = ""
+	parallelRequests = ""
```
Comment (leecalcote) [inline mesheryctl/cmd/perf.go]: @chandresh-pancholi in order to provide consistency of user experience, we'll want to mimic the language found in the UI here in the CLI. I believe we'll want this parameter to be named "concurrentRequests". @peterj @ashiscs did we converge on "concurrentRequests"?

@chandresh-pancholi great job in following the design outlined in our [cli doc](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.45f30knxz69z).

File: mesheryctl/cmd/perf.go [hunk 2]
```diff
@@ -123,14 +123,14 @@ var perfCmd = &cobra.Command{
...
+	perfCmd.Flags().StringVar(&testMesh, "mesh", "", "(optional) Name of the service mesh.")
 	// perfCmd.Flags().StringVar(&testFile, "file", "", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&qps, "qps", "0", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&concurrentRequests, "concurrent-requests", "1", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&testDuration, "duration", "30s", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&testCookie, "cookie", "meshery-provider=Local (ephemeral session) (free use)", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&loadGenerator, "load-generator", "fortio", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&qps, "qps", "1", "(optional) Queries per second")
+	perfCmd.Flags().StringVar(&parallelRequests, "parallel-requests", "1", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&testDuration, "duration", "10s", "(optional) Duration of the test like 10s, 5m, 2h. We are following the convention described at https://golang.org/pkg/time/#ParseDuration")
```
Comment (leecalcote) [inline mesheryctl/cmd/perf.go]: I believe that we'll want a default of 30s here.

File: mesheryctl/cmd/perf.go [hunk 3]
```diff
@@ -123,14 +123,14 @@ var perfCmd = &cobra.Command{
...
+	perfCmd.Flags().StringVar(&testURL, "url", "", "(required) URL of the endpoint to use for the test")
+	perfCmd.Flags().StringVar(&testName, "name", StringWithCharset(8), "(optional) A memorable name for the test.")
+	perfCmd.Flags().StringVar(&testMesh, "mesh", "", "(optional) Name of the service mesh.")
 	// perfCmd.Flags().StringVar(&testFile, "file", "", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&qps, "qps", "0", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&concurrentRequests, "concurrent-requests", "1", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&testDuration, "duration", "30s", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&testCookie, "cookie", "meshery-provider=Local (ephemeral session) (free use)", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&loadGenerator, "load-generator", "fortio", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&qps, "qps", "1", "(optional) Queries per second")
```
Comment (leecalcote) [inline mesheryctl/cmd/perf.go]: I believe we'll want a default of "0", which cause the behavior of "as many as possible".

File: mesheryctl/cmd/perf.go (hunk 1 above)
Comment (ashiscs) [reply mesheryctl/cmd/perf.go]: Yes sir we continued with concurrentRequests!!

**Comment by chandresh-pancholi** [comment]
Please update this document as I got the defaults and flags values from here.

**Comment by ashiscs** [comment]
Updated the document!!

**Comment by chandresh-pancholi** [comment]
Please merge if it Looks good to you.

File: mesheryctl/cmd/perf.go [hunk 4]
```diff
@@ -123,14 +123,14 @@ var perfCmd = &cobra.Command{
...
+	perfCmd.Flags().StringVar(&testName, "name", StringWithCharset(8), "(optional) A memorable name for the test.")
+	perfCmd.Flags().StringVar(&testMesh, "mesh", "", "(optional) Name of the service mesh.")
 	// perfCmd.Flags().StringVar(&testFile, "file", "", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&qps, "qps", "0", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&concurrentRequests, "concurrent-requests", "1", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&testDuration, "duration", "30s", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&testCookie, "cookie", "meshery-provider=Local (ephemeral session) (free use)", "DESCRIPTION")
-	perfCmd.Flags().StringVar(&loadGenerator, "load-generator", "fortio", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&qps, "qps", "0", "(optional) Queries per second")
+	perfCmd.Flags().StringVar(&concurrentRequests, "parallel-requests", "1", "DESCRIPTION")
```
Comment (ashiscs) [inline mesheryctl/cmd/perf.go]: @chandresh-pancholi here it should be (concurrent-requests) instead of (parallel-requests)..

Comment (chandresh-pancholi) [reply mesheryctl/cmd/perf.go]: oh crap.

**Comment by ashiscs** [comment]
@leecalcote sir, now this is ready to be merged!!

**Comment by leecalcote** [comment]
Fantastic. Thank you, gents. 👍

**Pull request merged**